app/scripts/lib/transaction/eip5792.ts

```diff
@@ -199,6 +210,14 @@ function validateUserDisabled(
   }
 }
 
+function validateUserPreference(dismissSmartAccountSuggestionEnabled: boolean) {
+  if (dismissSmartAccountSuggestionEnabled) {
+    throw rpcErrors.transactionRejected(
+      'User does not want to update account to smart account.',
```

Comment:
Could we combine this with the above to use the same error, as they're ultimately both preferences to not upgrade?

Reply:
Sure let me do this in following PR.

Reply:
Even so, from the perspective of the dApp, it's simply not supported for that chain and account. Plus I think we want to stick with `methodNotSupported` for now since the user hasn't technically rejected the transaction, but the act of upgrading.

Reply:
👍 I will implement code changes in following PR.

Reply:
I made these changes in PR while updating it to resolve conflicts